lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -232,8 +244,18 @@ class UntraceableObjectError(TracingError):
     """Raised when an object you're attempting to instrument cannot be autoinstrumented."""
 
 
-def _get_tracing_endpoint(tracing_endpoint_getter, self, charm):
-    if isinstance(tracing_endpoint_getter, property):
+class TLSError(TracingError):
+    """Raised when the tracing endpoint is https but we don't have a cert yet."""
+
+
+def _get_tracing_endpoint(
+    tracing_endpoint_getter: Union[_GetterType, _InstanceAttribute],
+    self: ops.CharmBase,
+    charm: Type[ops.CharmBase],
```

Comment:
since `self` is by convention usually meaning *this object* in python, maybe name it differently?  `charm_instance` and `charm_class` could work.

Reply:
I'm keeping it in the overridden charm init though, it makes it easier to reason about what's going on if you think from the perspective of the charm instance